tgui wires; refactor wire datums; more to come

neersighted opened the pull request:
:cl:
experiment: Refactor wires; port wire interface to tgui.
add: Wire colors are now fully randomized.
del: Remove pizza bombs, as the code and sprites were terrible.
add: Add pizza bomb cores, which can be combined with a pizza box to make a pizza bomb.
/:cl:

Fixes #14749
Fixes #12301

![2016-01-20_08-43-31](https://cloud.githubusercontent.com/assets/1049222/12461260/96173bca-bf7d-11e5-95a7-4d071f274509.png)
![2016-01-20_18-25-35](https://cloud.githubusercontent.com/assets/1049222/12467840/7afefd10-bfa4-11e5-8481-f6c0ed3d7676.gif)
![2016-01-20_17-31-39](https://cloud.githubusercontent.com/assets/1049222/12467838/7a6429e8-bfa4-11e5-9261-33de447559cd.png)
![image](https://cloud.githubusercontent.com/assets/1049222/12500084/6a5fd2fa-c075-11e5-9469-2e5dd1ac09bd.png)

AnturK committed:
tgui wires initial

neersighted committed:
Refactor wire datums

Finish wire port and refactor wire datums to use string keys; widen color
range to support more wires and more colors.

GunHog commented:
He just keeps going!!

neersighted committed:
Fix some small bugs in wire refactor; add all the colors!

neersighted committed:
Add better pizza bomb sprites

Iamgoofball commented:
@GunHog wires have been on the todo list for awhile because we have plans for the wire UI

optimumtact commented:
Plus those things mentioned are quite significant overhauls

also plus having lots of simple ui's makes it easier for newbs to look at them to figure out how it works

neersighted commented:
Yeah @GunHog a lot of things require significant refactors or total rewrites; and a lot of UIs are interdependent (see records consoles and PDAs). I plan to remove and rewrite PDAs from scratch; thus, big UIs will likely be post-freeze.

neersighted committed:
Refactor pizza_bomb, tweak pizza_bomb sprites

Thanks to @WJohn for the pizza bomb core sprites, and updated pizza box sprites

Removes the pizza_bomb and replaces it with pizzabox/bomb. Adds a special bomb core
for pizza bombs (a pizza bomb core can be used on pizza boxes and regular bombs, but
regular bomb cores cannotbe used for pizza bombs), and tweaks wire code a little bit.

neersighted committed:
Don't send duplicate JSON down the wire

neersighted committed:
Allow custom pizzas in pizza boxes

Also fix boxes icons being wrong after surviving an explosion

neersighted commented:
@WJohn thanks again for the prompt sprites

JJRcop commented:
:snowboarder: 
You're surfing on the freeze using tgui, I think the pizza box thing violates

neersighted commented:
@JJRcop it maintains existing functionality; doesn't add anything new (except allowing custom pizzas which is a bug); just factors out the explosive into a bomb core... The bomb core is just there for code purposes primarily. Before, if you tried to remove the core from a pizza bomb you got a regular bomb core, is this is an improvement.

JJRcop commented:
Well of course I'm no where near the arbiter of this stuff, I just wanted you to make sure you watch the lines, it seems like you're getting a little bit closer with this one.

neersighted committed:
Make wires use defines, put a wire var on /atom, fix tgui ui_host runtimes

neersighted committed:
Rewrite cryo

Similar balance but now healing is all in the reagent; and code is much simplified
Remove cryo leaking air, it just consumes O2 now. Gasses with a higher
specific_heat can be used to improve efficiency.
Greatly buffs cryo upgrades, in return for slight decrease in un-upgraded
performance due to atmos simulation.

Fix canister relabeling icons (Fixes #14742)

Add wrench sound to portable_atmospherics (consistency with ATMOSPHERICS/components)

Fix blood directly setting oxyloss (now uses adjustOxyLoss, balanced to be ~same as before)

neersighted committed:
Fix cryo layers

neersighted committed:
Rewrite heaters/freezers; port to tgui

They are now one type, thermomachine.

neersighted committed:
Re-structure tgui's ui_act

neersighted committed:
Fix meta heaters

neersighted committed:
Remove roomfiller canister, repath n20 to n2o

neersighted committed:
Fix conflicts

neersighted committed:
Prefer += to .Add()

For @RemieRichards

neersighted committed:
Fix colon cancer

MrStonedOne commented on code/modules/assembly/doorcontrol.dm:
```diff
@@ -47,15 +49,14 @@
-					if(D)
-						if(D.density)	D.open()
-						else			D.close()
-					return
```
This caused a bug fixed by https://github.com/tgstation/tgstation/pull/21125

@neersighted please don't remove spawns for no reason.
